packages/permission-controller/src/Permission.ts

```diff
@@ -349,6 +352,12 @@ export type PermissionValidatorConstraint = (
   target?: string,
 ) => void;
 
+export type PermissionSideEffect = {
+  onPermitted: () => void;
```

Comment:
If these are to be async, we'll want to return `Promise<void>` instead I support